#!/usr/bin/env python
# http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
# http://wiki.ros.org/dynamic_reconfigure/Tutorials/SettingUpDynamicReconfigureForANode%28cpp%29
PACKAGE = "commissioning_robot"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("roll_p", double_t, 0, "A double parameter", 0.002, -0.01, 0.01)
gen.add("roll_i", double_t, 0, "A double parameter", 0.000001, -0.00001 , 0.00001)
gen.add("roll_d", double_t, 0, "A double parameter", 0, -0.01, 0.01)
gen.add("pitch_p", double_t, 0, "A double parameter", 0.002, -0.01, 0.01)
gen.add("pitch_i", double_t, 0, "A double parameter", 0.000001, -0.00001 , 0.00001)
gen.add("pitch_d", double_t, 0, "A double parameter", 0, -0.01, 0.01)
gen.add("robot_vel_x", double_t, 0, "A double parameter", 0.5, 0, 1)
gen.add("robot_vel_y", double_t, 0, "A double parameter", 0.5, 0, 1)
gen.add("robot_vel_z", double_t, 0, "A double parameter", 0.5, 0, 1)

exit(gen.generate(PACKAGE, "commissioning_robot", "robot_param"))